Hyper/Scene.cs

```diff
@@ -53,27 +50,32 @@ public Scene(float aspectRatio)
         LightSources = GetLightSources(chunksPerSide, scalarFieldGenerator.AvgElevation);
         Projectiles = new List<Projectile>();
 
-        var bufferPool = new BufferPool();
-        Properties = new CollidableProperty<SimulationProperties>();
-        _characterControllers = new CharacterControllers(bufferPool);
-
-        SimulationManager = new SimulationManager<NarrowPhaseCallbacks, PoseIntegratorCallbacks>(new NarrowPhaseCallbacks(_characterControllers, Properties),
+        SimulationMembers = new Dictionary<BodyHandle, ISimulationMember>();
+        SimulationManager = new SimulationManager<PoseIntegratorCallbacks>(
             new PoseIntegratorCallbacks(new System.Numerics.Vector3(0, -10, 0)),
-            new SolveDescription(6, 1), bufferPool);
+            new SolveDescription(6, 1));
 
         var characterInitialPosition = new Vector3(0, scalarFieldGenerator.AvgElevation + 8, 15);
         Player = new Player.Player(CreatePhysicalHumanoid(characterInitialPosition));
+        SimulationMembers.Add(Player.BodyHandle, Player);
+        SimulationManager.RegisterContactCallback(Player.BodyHandle, contactInfo => Player.ContactCallback(contactInfo, SimulationMembers));
 
         int botsCount = 3;
         Bots = Enumerable.Range(0, botsCount) // initialize them however you like
             .Select(i => new Vector3(i * 4 - botsCount * 2, scalarFieldGenerator.AvgElevation + 5, i * 4 - botsCount * 2))
-            .Select(pos => new Humanoid(CreatePhysicalHumanoid(pos)))
+            .Select(pos =>
+            {
+                var humanoid = new Humanoid(CreatePhysicalHumanoid(pos));
```

Comment:
Consider putting these:
SimulationMembers.Add(humanoid.BodyHandle, humanoid);
                SimulationManager.RegisterContactCallback(humanoid.BodyHandle, contactInfo => humanoid.ContactCallback(contactInfo, SimulationMembers));
in the humanoid constructor

Reply:
Let me know if that's satisfactory

Reply:
I wanted to leave the decision about introducing a particular bot into the "contact callbacks world" up to the client code but I can change it if you think that this is unnecessary

Reply:
I think that the default should be adding it to the simulation. I cannot think of a reason why you would not want to always do that but its not a big deal so I'm leaving it up to you